website/content/en/highlights/2022-03-21-internal-schema-beta.md

```diff
@@ -0,0 +1,247 @@
+---
+date: "2022-03-21"
+title: "Beta release of internal schema support"
+description: "increasing correctness and ergonomics"
+authors: ["JeanMertz", "barieom"]
+pr_numbers: [11300]
+release: "0.21.0"
+hide_on_release_notes: false
+badges:
+  type: "new feature"
+  domains: ["schemas", "vrl"]
+  sources: ["datadog_agent"]
+  transforms: ["remap"]
+  sinks: ["datadog_logs"]
+---
+
+
+We're thrilled to announce Vector's initial support for log schema, providing
+type safety to your pipeline end-to-end for `datadog_agent` source and
+`datadog_logs` sink; this early beta release feature will increase your runtime
+correctness and improves VRL ergonomics.
+
+Without log schema support, Vector users previously lacked interoperability from
+a log integrations perspective. To integrate two systems, for example a
+`splunk_hec` source to a `datadog_logs` sink, Vector users had to manually
+intervene, requiring data mapping or transformations, as Vector could not map
+fields across a matrix of sources and sinks. Outside the overhead of manual
+intervention, this added a risk of fields being misplaced or duplicated in a
+downstream service, an issue that would only get discovered later in the
+development process.
+
+In addition, the lack of schema support caused considerable friction from a
+developer experience perspective. To expand, Vector users took advantage of VRL's
+infallibility by ensuring type safety through repeated calls to [coerce functions][coerce functions],
+such as `to_string`. However, each call to a coercion function necessitates
+error handling, which can be quite a burden.
+
+Most importantly, Vector users did not have visibility into whether malformed
+data is sent by an upstream client sending data to Vector. When a log with
+malformed schema enters a given Vector pipeline and causes an error, admin
+of the pipeline would need to intervene to find that error, as Vector was
+unable to provide validations errors on the edge.
+
+With this new release, Vector's log schema support will improve your developer
+experience and adding real-world reliability by guaranteeing end-to-end type
+safety. In short, any arbitrary log data that gets sent to a Vector pipeline will
+need to match a given schema. In this context, we define log schemas as the
+internal knowledge of data types — i.e. metadata — that informs features that
+require type information. This initial log schema support is limited to
+`datadog_agent` source and `datadog_logs` sink.
+
+This works by Vector implying a schema of some kind from the log source, then
+enforcing a schema based on the requirements of the sink. On the source side,
+there are two ways that Vector will imply a log schema: sources with underlying
```

Comment:
The way this is worded is confusing me, did you mean to separate the two like "sources with underlying protocols, and those with chosen codecs with unknown fields" or something along those lines?